betty/service.py

```diff
@@ -0,0 +1,127 @@
+"""
+Provide a service container.
+"""
+from collections import defaultdict
+from collections.abc import Callable, MutableMapping, Mapping, Awaitable, MutableSequence
+from types import TracebackType
+from typing import TypeVar, override, TypeAlias, Any, Self
+
+from betty.concurrent import AsynchronizedLock, _Lock
+
+ServiceT = TypeVar('ServiceT')
+NormalizedServiceId: TypeAlias = tuple[type[ServiceT], str | None]
```

Comment:
If we are to make services accessible from templates, then they MUST have unique string IDs.